test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -0,0 +1,1061 @@
+		operatorAction  *OperatorActionInfo
+		operatorCatalog *catalogd.Catalog
+		operator        *operatorv1alpha1.Operator
+		err             error
+	)
+	BeforeEach(func() {
+		sdkInfo = &SdkProjectInfo{
+			projectName: "plain-example",
+			domainName:  "plain.com",
+			group:       "cache",
+			version:     "v1alpha1",
+			kind:        "Memcached1",
+		}
+		bundleInfo = &BundleInfo{
+			baseFolderPath: "../../testdata/bundles/plain-v0",
+			bundles: []BundleContent{
+				{
+					bundleVersion: "0.1.0",
+				},
+				{
+					bundleVersion: "0.1.1",
+				},
+			},
+		}
+		catalogDInfo = &CatalogDInfo{
+			baseFolderPath:     "../../testdata/catalogs",
+			fbcFileName:        "catalog.yaml",
+			operatorName:       "plain-operator",
+			desiredChannelName: "beta",
+		}
+		operatorAction = &OperatorActionInfo{
+			installVersion: "0.1.0",
+			upgradeVersion: "0.1.1",
+		}
+		for i, b := range bundleInfo.bundles {
+			bundleInfo.bundles[i].bInputDir = catalogDInfo.operatorName + ".v" + b.bundleVersion
+			bundleInfo.bundles[i].imageRef = remoteRegistryRepo + catalogDInfo.operatorName + "-bundle:v" + b.bundleVersion
+		}
+		catalogDInfo.catalogDir = catalogDInfo.operatorName + "-catalog"
+		catalogDInfo.imageRef = remoteRegistryRepo + catalogDInfo.catalogDir + ":test"
+	})
+	It("should succeed", func() {
+		By("creating a new operator-sdk project")
+		err = sdkInitialize(sdkInfo)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("creating a new api and controller")
+		err = sdkNewAPIAndController(sdkInfo)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("generating CRD manifests")
+		err = sdkGenerateManifests(sdkInfo)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("generating bundle directory using kustomize")
+		// Creates bundle structure for the specified bundle versions
+		// Bundle content is same for the bundles now
+		for _, b := range bundleInfo.bundles {
+			err = kustomizeGenPlainBundleDirectory(sdkInfo, bundleInfo.baseFolderPath, b)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		By("building/pushing/kind loading bundle images from bundle directories")
+		for _, b := range bundleInfo.bundles {
+			dockerContext := filepath.Join(bundleInfo.baseFolderPath, b.bInputDir)
+			dockerfilePath := filepath.Join(dockerContext, "plainbundle.Dockerfile")
+			err = buildPushLoadContainer(b.imageRef, dockerfilePath, dockerContext, kindServer, GinkgoWriter)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		By("generating catalog directory by forming FBC and dockerfile using custom function")
+		imageRefsBundleVersions := make(map[string]string)
+		for _, b := range bundleInfo.bundles {
+			imageRefsBundleVersions[b.imageRef] = b.bundleVersion
+		}
+		err = genPlainCatalogDirectory(catalogDInfo, imageRefsBundleVersions)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("building/pushing/kind loading the catalog images")
+		dockerContext := catalogDInfo.baseFolderPath
+		dockerfilePath := filepath.Join(dockerContext, catalogDInfo.catalogDir+".Dockerfile")
```

Comment:
If the catalog name is `test-catalog`, the expectation here is to  form the `Dockerfile` name as `test-catalog.Dockerfile`. Hence concatenation is used.  This is to distinguish between various Dockerfiles formed. I think we can't achieve this if we use `filepath.Join` as suggested above.

Reply:
Ahhh, I see now. I was under the impression that it will be something like: `/somePathToDockerContext/catalogDir/.Dockerfile`, but you want it to be `/somePathToDockerContext/catalog-name.Dockerfile`. Got it now!

I think I just got confused by field name `catalogDir`: I asumed that this is not a catalog name, but part of path to catalog directory. In this case, as far as I understand - it is both.

I think to reduce confusion a bit we can use `fmt.Sprintf("%s.Dockerfile", catalogDInfo.catalogDir)` instead of `catalogDInfo.catalogDir+".Dockerfile")`.